internal/services/compute/virtual_machine_data_disk_attachment_resource.go

```diff
@@ -154,6 +197,15 @@ func resourceVirtualMachineDataDiskAttachmentCreateUpdate(d *pluginsdk.ResourceD
 		WriteAcceleratorEnabled: pointer.To(writeAcceleratorEnabled),
 	}
 
+	// there are ways to provision a VM without a StorageProfile and/or DataDisks
```

Comment:
This should also include a test to confirm that the existing VM won't have any diffs when a disk is attached this way

Reply:
1. I have re-run all existing testcases which are passed and attached in the [description](https://github.com/hashicorp/terraform-provider-azurerm/pull/25537#issue-2232455538).
2. Though this [comment ](https://github.com/hashicorp/terraform-provider-azurerm/pull/25537#discussion_r1557090305) said there are ways to provision a VM without a StorageProfile and/or DataDisks with certain types of images, I do not really know which image will trigger such scenario. Kindly let me know if you know the image type and I will add the testcase. However, this part of code will not cause any diff because this code only helps initialize the storage and disk when they are `nil`.